src/gcp/proto.ts

```diff
@@ -123,3 +123,37 @@ function fieldMasksHelper(
     fieldMasksHelper(newPrefixes, value, doNotRecurseIn, masks);
   }
 }
+
+/**
+ * Formats a single invoker to be used with IAM API calls, a vaild invoker is
+ * 'public', 'service-account', 'service-account@', or
+ * 'service-account@project.iam.gserviceaccount.com'.
+ * @param invokerMember the single invoker that is a member of the invoker array
+ * @param projectId the ID of the current project
+ * @returns a correctly formatted service account string
+ *
+ * @throws {@link Error} if the supplied service account string is empty
+ */
+export function formatInvokerMember(invokerMember: string, projectId: string): string {
```

Comment:
nit: I liked the method better when it was just for picking <sa>@<domain> vs <sa>@. Now that we special-case "public" I'm not sure if it works in other cases like the function.serviceAccount annotation. Can we keep this the way it was (formatServiceAccount), make the existing code use the helper, and then special case "public" => "allUsers" either in a helper function that uses formatServiceAccount or in setInvoker?

Reply:
Sure i'll change it back

Reply:
Changed back and added a helper function to get the correct invoker members.